1.5/Source/PawnStorages/PawnStorages/Farm/CompFarmNutrition.cs

```diff
@@ -101,16 +105,68 @@ public override void CompTick()
 
-                    if (pawn.TryGetComp(out CompEggLayer compLayer) && pawn.gender != Gender.Male)
+                    healthyPawns.Add(pawn);
+
+                    if (Props.doesProduction)
                     {
-                        EggLayerTick(compLayer, Props.animalTickInterval);
+                        if (pawn.TryGetComp(out CompEggLayer compLayer) && pawn.gender != Gender.Male)
+                        {
+                            EggLayerTick(compLayer, Props.animalTickInterval);
+                        }
+
+                        if (pawn.TryGetComp(out CompHasGatherableBodyResource compGatherable) &&
+                            pawn.gender != Gender.Male)
+                        {
+                            GatherableTick(compGatherable, Props.animalTickInterval);
+                        }
                     }
+                }
+
+                if (Props.doesBreeding)
+                {
+                    var types = from p in healthyPawns
+                        group p by p.kindDef
+                        into def
+                        select def;
 
-                    if (pawn.TryGetComp(out CompHasGatherableBodyResource compGatherable) && pawn.gender != Gender.Male)
+                    foreach (var type in types)
                     {
-                        GatherableTick(compGatherable, Props.animalTickInterval);
+                        var males = type.Where(p => p.gender == Gender.Male).ToList();
+                        var females = type.Where(p => p.gender != Gender.Male).ToList();
+
+                        if (!females.Any())
+                        {
+                            // no more females, reset
+                            breedingProgress[type.Key] = 0f;
+                        }
+
+                        // no males, stop progress
+                        if(!males.Any()) { continue; }
+
+                        var gestationTicks = AnimalProductionUtility.GestationDaysEach(type.Key.race) * 60000 * Props.breedingScaler;
+
+                        var progressPerCycle = Props.animalTickInterval/ gestationTicks;
+
+
+                        breedingProgress.TryAdd(type.Key, 0.0f);
+                        
+                        breedingProgress[type.Key] = Mathf.Clamp(breedingProgress[type.Key] + progressPerCycle * females.Count, 0f, 1f);
+
+                        if (!(breedingProgress[type.Key] >= 1f)) continue;
+                        breedingProgress[type.Key] = 0f;
+                        var newPawn = PawnGenerator.GeneratePawn(
+                            new PawnGenerationRequest(
+                                type.Key,
+                                Faction.OfPlayer,
+                                allowDowned: true,
+                                forceNoIdeo: true, 
+                                developmentalStages: DevelopmentalStage.Newborn
+                            ));
+
+                        daysProduce.Add(newPawn);
```

Comment:
I've set it so that the breeding farms don't produce - breeding only, so no worries on the production front. 
That said, if we want to also hold them until grown that's something to add. 

Would it make sense to add them to the main storage, or have a separate storage for the baby animals?

What behaviour on storage full? Stop creating new, eject new ones?

Reply:
Add them to the main storage and stop breeding if full imo. Stops breeding getting super out of control for little effort.

It's just the question of will they ever age up in storage to start contributing or come out still age 0

Reply:
I think I'll probably need to pull out the logic from `Pawn_AgeTracker.AgeTick()` and run it at the scaled ticks. Shouldn't be too difficult, main thing is to ensure the bio age ticks up correctly.